### ISSUE TITLE
User Guide: Clarify upgrade path for silent installations

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#11618]
### Is your feature request related to a problem? Please describe.

Section 16.2 of the User Guide lists all command-line switches that NVDA will accept; one of these is "--install-silent", which allows NVDA to be installed to a machine without any manual intervention; this is really useful in large organisations, or where automation is required. What isn't overly clear in the User Guide, however, is whether it's possible to use this switch to perform an upgrade of NVDA - or whether this needs to either be performed by the end-user (when/if required), or whether the current install must first be removed and then NVDA reinstalled on the machine in question. This information would be useful to have for administrators to have when planning a large deployment of NVDA.

**Comment by lukaszgo1** [comment]
Since "--install-silent" behaves exactly like the manual install it is reasonable to assume that it simply upgrades the existing installation. I'd argue that if it would behave otherwise then we should document it, but since the behavior is an expected one there is nothing to document. The think which I'm not entirely clear about however is how the silent install would behave in case of a downgrade, and that may indeed be tested and documented.

**Comment by elliott94** [comment]
The other question that's just come to me relates to how NVDA behaves when this switch is activated when it's already running: is any kind of notification given to the user when it's about to be upgraded, or is it simply killed, upgraded, and then restarted automatically after the process completes? Or, does the user instead have to manually launch it? Is there an easy way for them to know when the upgrade has completed? Whilst some of this can be up to a system administrator to decide, further documentation in this area would be useful I think. Maybe we need to develop a "System Administrator's Guide"... but that's for a different discussion! :)

**Comment by feerrenrut** [comment]
I don't think it would do any harm to explicitly note something like this. It's not possible for someone who is unfamiliar to know if the absence of documentation means it works how they expect, or in some surprising way that merely isn't documented.

I also think a Sys-admin guide is a good idea. Given the number of issues you have created on this topic @elliott94, would you be interested in this project?

**Comment by elliott94** [comment]
Sure - these questions/improvements are all things that I've come across over the last several weeks, so more than happy to document them to help users/administrators in a similar situation in the future. Just several things of note, however:

Whilst I'm more than happy to collect all of these pieces of information together and put them into some kind of guide, lack of time right now means that I may not be able to get overly familiar with a development environment - if I created something along the lines of an HTML file, for example and added appropriate headings/links to reference materials, would somebody then be able to put this in the proper format and push it to the main repo for me? I'd really appreciate this. :)

Thank you.

**Comment by feerrenrut** [comment]
I think that would be a great start @elliott94. Plain text / markdown would be a good place to start, what will require minimal changes to convert to the right format. Even just as a comment on the issues.

**Comment by elliott94** [comment]
No problem at all. Just a couple of things:

Would you like me to create a new GitHub issue that's dedicated specifically to the Admin Guide, and keep this one exclusively about clarification of the upgrade path for NVDA? I can close this when relevant information has been added to the Admin Guide.

If at all possible, it'd be really useful to get some developer clarifications on content that I was looking to possibly add. What would be the easiest way of obtaining this - shall I simply post a list of questions that I've got, or would you prefer me to use a different approach? My questions mainly relate to NVDA internals that I'm not familiar with, but I feel would be good to clarify for the guide.

Finally - I was planning on adding info to the guide on command line switches and registry locations - both of which are currently covered in the main User Guide. Would we plan to keep this info in the User Guide, or remove it and instead make reference to the Admin Guide so that information can be maintained in a single location?

Thank you.
